public/scripts/project/lock.js

```diff
@@ -0,0 +1,51 @@
+// Our lock system for projects is very simple. If the lock exists,
+// it's almost guaranteed to be valid since the expiry serves
+// to prevent completely locking access to a project in edge cases.
+define(function() {
+  var EXPIRY = 60000; // 1000ms * 60s
```

Comment:
I think 30s should be fine! I don't think we want to write to local storage every 10s when we have more than one project open. 7 projects open in different tabs sending commands to localStorage seems like overkill. I think a slightly more relaxed time interval would be better, especially since I don't think we should consider browser crashes as something that is very easy to recover from.